pkg/sidecar/k8s_instance.go

```diff
@@ -0,0 +1,386 @@
+			}
+
+			route.Dst.IP = gw
+			route.Dst.Mask = net.CIDRMask(32, 32)
+
+			if err := netlinkHandle.RouteAdd(&route); err != nil {
+				return nil, fmt.Errorf("failed to add route while restricting gw route: %v", err)
+			}
+		}
+
+		if route.Dst == nil && route.Src == nil {
+			if err := netlinkHandle.RouteDel(&route); err != nil {
+				return nil, fmt.Errorf("failed to drop default route: %v", err)
+			}
+		}
+	}
+
+	// Add specific routes to redis if redis uses this link.
+	if err := netlinkHandle.RouteAdd(&redisRoute); err != nil {
+		return nil, fmt.Errorf("failed to add redis route: %v", err)
+	}
+
+	return NewInstance(runenv, info.Config.Hostname, network)
+}
+
+type k8sLink struct {
+	*NetlinkLink
+	IPv4, IPv6 *net.IPNet
+
+	rt      *libcni.RuntimeConf
+	netconf *libcni.NetworkConfigList
+}
+
+type K8sNetwork struct {
+	container   *dockermanager.Container
+	activeLinks map[string]*k8sLink
+	nl          *netlink.Handle
+	cninet      *libcni.CNIConfig
+	subnet      string
+	netnsPath   string
+}
+
+func (n *K8sNetwork) Close() error {
+	return nil
+}
+
+func (n *K8sNetwork) ConfigureNetwork(ctx context.Context, cfg *sync.NetworkConfig) error {
+	if cfg.Network != "default" {
+		return errors.New("configured network is not default")
+	}
+
+	link, online := n.activeLinks[cfg.Network]
+
+	// Are we _disabling_ the network?
+	if !cfg.Enable {
+		// Yes, is it already disabled?
+		if online {
+			// No. Disconnect.
+			if err := n.cninet.DelNetworkList(ctx, link.netconf, link.rt); err != nil {
+				return fmt.Errorf("when 6: %w", err)
+			}
+			delete(n.activeLinks, cfg.Network)
+		}
+		return nil
+	}
+
+	if online && ((cfg.IPv6 != nil && !link.IPv6.IP.Equal(cfg.IPv6.IP)) ||
+		(cfg.IPv4 != nil && !link.IPv4.IP.Equal(cfg.IPv4.IP))) {
+		// Disconnect and reconnect to change the IP addresses.
+		logging.S().Debug("disconnect and reconnect to change the IP addr", "cfg.IPv4", cfg.IPv4, "link.IPv4", link.IPv4.String(), "container", n.container.ID)
+		//
+		// NOTE: We probably don't need to do this on local docker.
+		// However, we probably do with swarm.
+		online = false
+		if err := n.cninet.DelNetworkList(ctx, link.netconf, link.rt); err != nil {
+			return fmt.Errorf("when 5: %w", err)
+		}
+		delete(n.activeLinks, cfg.Network)
+	}
+
+	// Are we _connected_ to the network.
+	if !online {
+		// No, we're not.
+		// Connect.
+		if cfg.IPv6 != nil {
+			return errors.New("ipv6 not supported")
+		}
+
+		var (
+			netconf *libcni.NetworkConfigList
+			err     error
+		)
+		if cfg.IPv4 == nil {
+			netconf, err = newNetworkConfigList("net", n.subnet)
+		} else {
+			netconf, err = newNetworkConfigList("ip", cfg.IPv4.String())
+		}
+		if err != nil {
+			return fmt.Errorf("failed to generate new network config list: %w", err)
+		}
+
+		cniArgs := [][2]string{}                   // empty
+		capabilityArgs := map[string]interface{}{} // empty
+		ifName := "eth1"
```

Comment:
Let's make constants for "eth1" and "eth2" (control network interface, data network interface).